mkosi/__init__.py

```diff
@@ -7469,11 +7470,6 @@ def check_root() -> None:
         die("Must be invoked as root.")
 
 
-def check_native(args: MkosiArgs) -> None:
-    if args.architecture is not None and args.architecture != platform.machine() and args.build_script:
-        die("Cannot (currently) override the architecture and run build commands")
```

Comment:
It will be a really long time before this lands in some distros and I think most people using mkosi in earnest will use the main branch (or at least a recent tag, although thanks to @bluca mkosi is really current in Debian now), so I think it would be safer to either guard this behind a distribution check or start checking the systemd version we are running on.

Reply:
yeah checking the nspawn version seems prudent

Reply:
Added nspawn version check